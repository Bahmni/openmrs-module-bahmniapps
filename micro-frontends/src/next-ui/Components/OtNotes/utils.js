import axios from 'axios';
import { OT_NOTES_BASE_URL } from "../../constants";

export const deleteOtNote = async (id) => {
    try {
        const response = await axios.delete(OT_NOTES_BASE_URL + "/" + id, {
            headers: {"Accept": "application/json", "Content-Type": "application/json", withCredentials: true}
        });
        return response.data;

    } catch (e) {
        console.error(e);
    }
}

export const constructPayload = function (noteDate, note, noteEndDate) {
    const payload = [];
    const currentDate = new Date(noteDate);
    while (currentDate <= noteEndDate) {
        payload.push({
            noteTypeName: "OT module",
            noteDate: new Date(currentDate),
            noteText: note
        });
        currentDate.setDate(currentDate.getDate() + 1);
    }
    return payload;
};

export const saveNote = async (note, startDate, endDate) => {
    try{
        const payload = endDate != null ? constructPayload(startDate, note, endDate) : [{
            noteTypeName: "OT module",
            noteDate: startDate,
            noteText: note
        }];
        const response = await axios.post(OT_NOTES_BASE_URL, payload);
        return response.data;
    }
    catch (e) {
        console.error(e);
    }
}

export const updateNoteForADay = async (noteId, note, providerUuid) => {
    const payload = {
        noteText: note,
        providerUuid: providerUuid
    };
    try{
        const response = await axios.post(OT_NOTES_BASE_URL + "/" + noteId, payload, {
            headers: {"Accept": "application/json", "Content-Type": "application/json"}
        });
        return response.data;
    }
    catch (e) {
        console.error(e);
    }
};