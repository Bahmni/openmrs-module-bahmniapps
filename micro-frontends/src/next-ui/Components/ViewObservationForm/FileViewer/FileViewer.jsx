import React from "react";
import propTypes from "prop-types";
import { Tile } from "carbon-components-react";
import { BuildFileViewer } from "./BuildFileViewer/BuildFileViewer";
import "./FileViewer.scss";

export const FileViewer = (props) => {
  const { members, isHeader = false } = props;
  const groupByConceptUuid = members.reduce((acc, member) => {
    acc[member.concept.uuid] = acc[member.concept.uuid] || [];
    acc[member.concept.uuid].push(member);
    return acc;
  }, {});

  return (
    <>
      {Object.values(groupByConceptUuid).map((elementList, index) => {
        return isHeader ? (
          <Tile key={index}>
            <div key={index} className="file-section">
              <span
                className="section-header row-label"
                data-testid={`section-label-${index}`}
              >
                {elementList[0].concept.shortName}
              </span>
              <span
                className="row-value viewer-value-block"
                key={`row-value-${index}`}
              >
                {elementList.map((subItem, index) => (
                  <BuildFileViewer item={subItem} index={index} />
                ))}
              </span>
            </div>
          </Tile>
        ) : (
          <>
            <div
              key={index}
              className="row-element"
              data-testid={`subItem-${index}`}
            >
              <div className="row-content">
                <div className="row-label">
                  {elementList[0].concept.shortName}
                </div>
                <div className="row-value viewer-value-block">
                  {elementList.map((subItem, index) => (
                    <BuildFileViewer item={subItem} index={index} />
                  ))}
                </div>
              </div>
            </div>
          </>
        );
      })}
    </>
  );
};

FileViewer.propTypes = {
  members: propTypes.array,
  isHeader: propTypes.bool,
};
export default FileViewer;
