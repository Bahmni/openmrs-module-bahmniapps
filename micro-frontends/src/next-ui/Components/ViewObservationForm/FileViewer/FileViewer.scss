.form-obs-image {
   height: 93px;
   width: 93px;
   border: 1px solid #ccc;

   &:hover {
      cursor: pointer;
      box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25), 0 10px 10px rgba(0, 0, 0, 0.22);
      border-color: #eee;
   }
}

.form-obs-video-image {
   height: 100%;
   width: 100%;
}

.viewer-value-block {
   display: flex;
   width: 100%;
   flex-wrap: wrap;
}

.image-viewer {
   display: flex;
   width: 50%;
   padding-bottom: 10px;
}

.video-viewer {
   display: flex;
   width: 100%;
   padding-bottom: 10px;
}

.file-notes-section {
   padding: 10px;
   width: 85%;
}

.provider-info {
   font-size: 12px;
   font-weight: bold;
}

.file-section {
   display: flex;
}

.download-link{
   display: flex;
}

.video-play-button {
   position: relative;
   height: 93px;
   width: 93px;
   display: inline-block;

   &:hover {
      cursor: pointer;
      filter: brightness(1.5);
      box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25), 0 10px 10px rgba(0, 0, 0, 0.22);
      border-color: #eee;
   }
}

.video-play-button-icon {
   position: absolute;
   top: 50%;
   left: 50%;
   transform: translate(-50%, -50%);
}