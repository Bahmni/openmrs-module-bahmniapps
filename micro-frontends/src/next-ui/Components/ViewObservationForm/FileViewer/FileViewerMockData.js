export const mockInfoMap = {
  providerName: "test provider",
  encounterDateTime: "encounterDateTime",
  comment: "comment",
  fileName: "fileName",
  fileType: "image/jpeg",
  fileSource: "fileSource",
};

export const mockVideoInfoMap = {
  providerName: "test provider",
  encounterDateTime: "encounterDateTime",
  comment: "comment",
  fileName: "fileName",
  fileType: "image/jpeg",
  fileSource: "fileSource",
};

export const mockIndex = 0;

export const mockItem = {
  providers: [
    {
      name: "test provider",
    },
  ],
  encounterDateTime: 1713856719000,
  comment: "comment",
  complexData: {
    title: "fileName",
    mimeType: "image/jpeg",
  },
  value: "fileSource",
};

export const mockPdfItem = {
  providers: [
    {
      name: "test provider",
    },
  ],
  encounterDateTime: 1713856719000,
  comment: "comment",
  complexData: {
    title: "fileName",
    mimeType: "application/pdf",
  },
  value: "fileSource",
};

export const mockVideoItem = {
  providers: [
    {
      name: "test provider",
    },
  ],
  encounterDateTime: 1713856719000,
  comment: "comment",
  complexData: {
    title: "fileName",
    mimeType: "video/mp4",
  },
  value: "fileSource",
};

export const observationList = [
  {
    encounterDateTime: 1713856719000,
    groupMembers: [],
    providers: [
      {
        uuid: "c1c26908-3f10-11e4-adec-0800271c1b75",
        name: "test provider one",
        encounterRoleUuid: "a0b03050-c99b-11e0-9572-0800200c9a66",
      },
    ],
    isAbnormal: null,
    duration: null,
    type: "Complex",
    encounterUuid: "276f3f45-2bdb-4166-b495-8e3463774034",
    interpretation: null,
    complexData: {
      title: "51196-Consultation-3c01db16-e63a-4b83-9159-dc3f4e8ebb01.mp4",
      mimeType: "video/mp4",
    },
    conceptNameToDisplay: "Patient Video",
    concept: {
      uuid: "eef8e969-269d-40c4-b7c9-182da11c2653",
      name: "Patient Video",
      dataType: "Complex",
      shortName: "Patient Video",
    },
    conceptUuid: "eef8e969-269d-40c4-b7c9-182da11c2653",
    comment: "Sample comments are added one",
    value: "51200/51196-Consultation-3c01db16-e63a-4b83-9159-dc3f4e8ebb01.mp4",
  },
  {
    encounterDateTime: 1713856719000,
    groupMembers: [],
    providers: [
      {
        uuid: "c1c26908-3f10-11e4-adec-0800271c1b75",
        name: "test provider one",
      },
    ],
    isAbnormal: null,
    type: "Complex",
    interpretation: null,
    complexData: {
      title: "51196-Consultation-37e2a799-b44c-412f-9a16-1f04ae8f9a34.mp4",
      mimeType: "video/mp4",
    },
    conceptNameToDisplay: "Patient Video",
    concept: {
      uuid: "eef8e969-269d-40c4-b7c9-182da11c2653",
      name: "Patient Video",
      dataType: "Complex",
      shortName: "Patient Video",
    },
    conceptUuid: "eef8e969-269d-40c4-b7c9-182da11c2653",
    comment: "Sample comments are added two",
    value: "51200/51196-Consultation-37e2a799-b44c-412f-9a16-1f04ae8f9a34.mp4",
  },
  {
    encounterDateTime: 1713856719000,
    groupMembers: [],
    providers: [
      {
        uuid: "c1c26908-3f10-11e4-adec-0800271c1b75",
        name: "test provider one",
      },
    ],
    isAbnormal: null,
    type: "Complex",
    interpretation: null,
    complexData: {
      title: "51196-Consultation-445dd7fc-a7c7-4e9c-8ea6-70a3f567d8d4.jpeg",
      mimeType: "image/jpeg",
    },
    conceptNameToDisplay: "Image",
    concept: {
      uuid: "6ca8ddf3-4e3d-4116-bd4e-a5147e580931",
      name: "Image",
      dataType: "Complex",
      shortName: "Image",
    },
    comment: "Sample comments are added three",
    value: "51200/51196-Consultation-445dd7fc-a7c7-4e9c-8ea6-70a3f567d8d4.jpeg",
  },
  {
    encounterDateTime: 1713856719000,
    groupMembers: [],
    providers: [
      {
        uuid: "c1c26908-3f10-11e4-adec-0800271c1b75",
        name: "test provider one",
      },
    ],
    isAbnormal: null,
    type: "Complex",
    interpretation: null,
    complexData: {
      title: "51196-Consultation-46b1cbfc-00a7-4cbb-afdf-71ac2b040182.jpeg",
      mimeType: "image/jpeg",
    },
    conceptNameToDisplay: "Image",
    concept: {
      uuid: "6ca8ddf3-4e3d-4116-bd4e-a5147e580931",
      name: "Image",
      dataType: "Complex",
      shortName: "Image",
    },
    comment: "Sample comments are added four",
    value: "51200/51196-Consultation-46b1cbfc-00a7-4cbb-afdf-71ac2b040182.jpeg",
  },
  {
    encounterDateTime: 1713856719000,
    groupMembers: [],
    providers: [
      {
        uuid: "c1c26908-3f10-11e4-adec-0800271c1b75",
        name: "test provider one",
      },
    ],
    isAbnormal: null,
    type: "Complex",
    interpretation: null,
    complexData: {
      title: "51196-Consultation-aa0a417a-0a61-4c97-b4be-fe3fc43ea371.pdf",
      mimeType: "application/pdf",
    },
    conceptNameToDisplay: "Image",
    concept: {
      uuid: "6ca8ddf3-4e3d-4116-bd4e-a5147e580931",
      name: "Image",
      dataType: "Complex",
      shortName: "Image",
    },
    uuid: "8a643679-fcb2-4246-9b10-01a3d1eb4eb6",
    comment: null,
    value: "51200/51196-Consultation-aa0a417a-0a61-4c97-b4be-fe3fc43ea371.pdf",
  },
];

export const observationListWithGroupMembers = [{
  "encounterDateTime": 1713953624000,
  "groupMembers": [
    {
      "encounterDateTime": 1713953624000,
      "groupMembers": [],
      "providers": [
        {
          "uuid": "c1c26908-3f10-11e4-adec-0800271c1b75",
          "name": "provider one",
        }
      ],
      "type": "Complex",
      "interpretation": null,
      "complexData": {
        "title": "51196-Consultation-e1e88b46-5db9-43c4-97fc-34096bdaad25.jpeg",
        "mimeType": "image/jpeg",
      },
      "conceptNameToDisplay": "Image 0-9-3",
      "valueAsString": "51200/51196-Consultation-e1e88b46-5db9-43c4-97fc-34096bdaad25.jpeg",
      "concept": {
        "uuid": "4fbf4b32-4411-436c-9c3b-1a907685a0a0",
        "name": "Image 0-9-3",
        "dataType": "Complex",
        "shortName": "Image 0-9-3",
      },
      "comment": "Notes 1",
      "value": "51200/51196-Consultation-e1e88b46-5db9-43c4-97fc-34096bdaad25.jpeg"
    },
    {
      "encounterDateTime": 1713953624000,
      "groupMembers": [],
      "providers": [
        {
          "uuid": "c1c26908-3f10-11e4-adec-0800271c1b75",
          "name": "provider one",
        }
      ],
      "type": "Complex",
      "interpretation": null,
      "complexData": {
        "title": "51196-Consultation-1cb9db39-987b-4f36-bf9b-3f4538e9b478.png",
        "mimeType": "image/png",
      },
      "conceptNameToDisplay": "Image 0-9-3",
      "valueAsString": "51200/51196-Consultation-1cb9db39-987b-4f36-bf9b-3f4538e9b478.png",
      "concept": {
        "uuid": "4fbf4b32-4411-436c-9c3b-1a907685a0a0",
        "name": "Image 0-9-3",
        "dataType": "Complex",
        "shortName": "Image 0-9-3",
      },
      "conceptUuid": "4fbf4b32-4411-436c-9c3b-1a907685a0a0",
      "comment": "Notes 2",
      "value": "51200/51196-Consultation-1cb9db39-987b-4f36-bf9b-3f4538e9b478.png"
    }
  ],
  "providers": [
    {
      "uuid": "c1c26908-3f10-11e4-adec-0800271c1b75",
      "name": "provider one",
    }
  ],
  "type": null,
  "interpretation": null,
  "conceptNameToDisplay": "Consultation Images",
  "concept": {
    "uuid": "c3ae29e3-9836-4df9-bcef-8d01b776854d",
    "name": "Consultation Images",
    "dataType": "N/A",
    "shortName": "Consultation Images",
  },
  "conceptUuid": "c3ae29e3-9836-4df9-bcef-8d01b776854d",
  "comment": null,
  "value": "51200/51196-Consultation-1cb9db39-987b-4f36-bf9b-3f4538e9b478.png, 51200/51196-Consultation-e1e88b46-5db9-43c4-97fc-34096bdaad25.jpeg"
}];