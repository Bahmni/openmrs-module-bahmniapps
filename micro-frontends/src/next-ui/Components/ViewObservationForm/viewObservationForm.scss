.view-observation-form-modal {
  .bx--modal-container {
    background: #f0f0f0;
    border-radius: 5px;
    width: 80%;
    height: 84%;
    max-width: 1180px;
    top: 96px;
  }

  .bx--modal-content {
    padding-bottom: 10px;
    margin-bottom: 35px;
    padding-top: 0;
  }

  .bx--tile {
    margin-bottom: 20px;
    min-height: auto;
    padding: 10px;
  }

  .content-body {
    border: 1px solid #669999;
    border-radius: 4px 4px 0 0;
    box-shadow: 0px 4px 4px 0px #0000004f;
  }

  .section-title {
    font-family: OpenSans, Arial, sans-serif;
    padding: 10px;
    margin-top: 0;
    margin-bottom: 0;
    color: #ffffff;
    font-size: 16px;
    clear: both;
    background: #669999;
    position: sticky;
    border-radius: 0;
    top: 0;
  }

  .section-body {
    padding: 20px 10px;
    background-color: #ffffff;
    color: #525252;
  }

  .section-header {
    font-family: IBM Plex Sans;
    font-size: 14px;
    font-weight: 500;
    line-height: 22px;
  }

  .row-body {
    background-color: #ffffff;
    font-size: 14px;
    margin-top: 5px;
  }

  .row-element {
    display: flex;
    flex-direction: column;
    padding: 10px 20px;
    border-bottom: 1px solid #c6c6c6;
  }

  .row-body div:last-child {
    border-bottom: none;
  }

  .row-content {
    display: flex;
  }

  .row-label {
    width: 30%;
    margin-right: 10px;
    word-break: break-word;
  }

  .row-value {
    width: 70%;
    white-space: pre-wrap;
    word-wrap: break-word;
  }

  .w-70 {
    width: 70%;
  }

  .sub-items-title {
    display: block;
    padding-top: 10px;
  }

  .sub-items-body {
    background: #f4f4f4;
    padding-bottom: 20px;
  }

  .notes-section {
    display: block;
    background: #d0e2ff;
    color: #525252;
    min-height: 40px;
    margin-top: 10px;
    font-size: 12px;
    font-style: italic;
    font-weight: 400;
    line-height: 17px;
    letter-spacing: 0em;
  }

  .document-icon {
    margin-right: 10px;
  }

  .sub-label {
    font-size: 11px;
  }

  .is-abnormal {
    color: red;
  }

  @media (min-width: 42rem) {
    .bx--modal-container {
      position: fixed;
    }
  }

  .print-button {
    position: fixed;
    z-index: 500;
    right: 30px;
    top: 40px;
  }
}
