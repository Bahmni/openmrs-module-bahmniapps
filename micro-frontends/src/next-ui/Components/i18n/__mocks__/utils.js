export function getTranslations() {
  return import("../../../../../public/i18n/locale_en.json").then(
    (module) => module.default
  );
}

export const getLocale = () => "en";
