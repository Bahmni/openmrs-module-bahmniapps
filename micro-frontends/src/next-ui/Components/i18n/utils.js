import { LS_LANG_KEY, BASE_URL } from "../../constants";

const translationsBaseUrl = "i18n";

export function getLocale() {
  return localStorage.getItem(LS_LANG_KEY) || "en";;
}

export const getTranslations = async (locale) => {
  const fileName = `locale_${locale}.json`;
  return fetchTranslations(fileName);
};

async function fetchTranslations(fileName) {
  const url = `${BASE_URL}${translationsBaseUrl}/${fileName}`;
  const response = await fetch(url);
  return response.json();
}
