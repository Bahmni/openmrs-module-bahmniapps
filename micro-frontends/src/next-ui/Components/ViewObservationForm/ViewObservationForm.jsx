import React from "react";
import { Modal, Tile, Loading } from "carbon-components-react";
import propTypes from "prop-types";
import { Document } from "@carbon/icons-react/next";
import TileItem from "./TileItem/TileItem";
import {
  subLabels,
  isAbnormal,
  getValue,
  isValidFileFormat,
} from "../../utils/FormDisplayControl/FormView";

import "./viewObservationForm.scss";
import { FileViewer } from "./FileViewer/FileViewer";

export const ViewObservationForm = (props) => {
  const {
    formName,
    formNameTranslations,
    closeViewObservationForm,
    formData,
    isViewFormLoading,
    showPrintOption,
    printForm,
  } = props;

  const imageItems = formData?.filter((member) => isValidFileFormat(member));

  return (
    <div>
      <Modal
        open
        passiveModal
        className="view-observation-form-modal"
        onRequestClose={closeViewObservationForm}
      >
        {showPrintOption && (
          <button className="confirm print-button" onClick={printForm}>
            Print
          </button>
        )}
        <section className="content-body">
          <h2 className="section-title">{formNameTranslations}</h2>
          {isViewFormLoading ? (
            <div>
              <Loading />
            </div>
          ) : (
            <section className="section-body">
              {imageItems?.length > 0 && (
                <FileViewer members={imageItems} isHeader={true} />
              )}
              {formData.map((section, index) => {
                if (!isValidFileFormat(section)) {
                  return (
                    <Tile key={index}>
                      <div
                        style={{
                          display: section?.groupMembers?.length
                            ? "block"
                            : "flex",
                        }}
                      >
                        <span
                          className={`section-header ${
                            section?.groupMembers?.length ? "" : "row-label"
                          } ${
                            isAbnormal(section.interpretation)
                              ? "is-abnormal"
                              : ""
                          }`}
                          data-testid={`section-label-${index}`}
                        >
                          {section.concept.shortName}&nbsp;
                          <span className="sub-label">
                            {subLabels(section.concept)}
                          </span>
                        </span>
                        {section?.groupMembers?.length ? (
                          <TileItem items={section.groupMembers} />
                        ) : (
                          <div
                            className={`row-value ${
                              isAbnormal(section.interpretation)
                                ? "is-abnormal"
                                : ""
                            }`}
                          >
                            {getValue(section)}
                            &nbsp;{section.concept.units || ""}
                          </div>
                        )}
                      </div>
                      {section.comment && (
                        <span className="notes-section">
                          <Document className="document-icon" />
                          {`${section.comment} - by ${
                            (section.providers && section.providers[0]?.name) ||
                            ""
                          }`}
                        </span>
                      )}
                    </Tile>
                  );
                }
              })}
            </section>
          )}
        </section>
      </Modal>
    </div>
  );
};

ViewObservationForm.propTypes = {
  formName: propTypes.string,
  formNameTranslations: propTypes.string,
  closeViewObservationForm: propTypes.func,
  formData: propTypes.array,
  isViewFormLoading: propTypes.bool,
  showPrintOption: propTypes.bool,
  printForm: propTypes.func
};
export default ViewObservationForm;
