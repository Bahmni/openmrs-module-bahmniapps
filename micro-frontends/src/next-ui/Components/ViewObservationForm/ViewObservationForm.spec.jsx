import React from "react";
import { render, screen, within } from "@testing-library/react";
import ViewObservationForm from "./ViewObservationForm.jsx";
import { observationList, observationListWithGroupMembers } from "./FileViewer/FileViewerMockData";
import { formatDate } from "../../utils/utils";

const initialProps = {
  formName: "Vitals",
  formNameTranslations: "Vitals",
  isViewFormLoading: false,
  formData: [
    {
      concept: {
        shortName: "Pulse",
        hiNormal: 100,
        lowNormal: 60,
      },
      value: "105",
      groupMembers: [],
      interpretation: "Abnormal",
      comment: "notes example",
      providers: [{ name: "test provider" }],
    },
    {
      concept: {
        shortName: "Pulse",
        hiNormal: 100,
        lowNormal: 60,
        units: "beats/min",
      },
      valueAsString: "105",
      groupMembers: [],
      comment: "notes example",
    },
    {
      concept: { shortName: "Blood Pressure" },
      groupMembers: [
        {
          concept: {
            shortName: "Systolic Blood Pressure",
            hiNormal: 140,
            lowNormal: 100,
            units: "mmHg"
          },
          value: "120",
          interpretation: "Abnormal",
          comment: "notes example",
          providers: [{ name: "test provider" }],
        },
        {
          concept: {
            shortName: "Diastolic Blood Pressure",
            hiNormal: 90,
            lowNormal: 60,
            units: "mmHg",
          },
          value: "80",
          comment: "notes example",
        },
      ],
    },
  ],
};

describe("ViewObservationForm", () => {
  it("should match the screenshot", () => {
    const { container } = render(<ViewObservationForm {...initialProps} />);
    expect(container).toMatchSnapshot();
  });

  it("should highlight member in red if it is abnormal", () => {
    render(<ViewObservationForm {...initialProps} />);
    const element = screen.getByTestId("section-label-0");
    expect(element.classList.contains("is-abnormal")).toBeTruthy();
  });

  it("should show loader", () => {
    const updatedProps = { ...initialProps, isViewFormLoading: true };
    render(<ViewObservationForm {...updatedProps} />);
    expect(screen.queryAllByText("Active loading indicator")).toHaveLength(2);
  });

  it("should render and group complex type like video/image/pdf which is in first level of hierarchy", () => {
    const updatedProps = { ...initialProps, formData: observationList};
    const { container } = render(<ViewObservationForm {...updatedProps} />);

    const fileSections = container.querySelectorAll(".file-section");
    expect(fileSections).toHaveLength(2);
    const video = fileSections[0];
    
    expect(video).not.toBeNull();
    expect(within(video).getByText(/Patient Video/i)).toBeTruthy();
    expect(within(video).getByText(/Sample comments are added one/i)).toBeTruthy();
    expect(within(video).getByText(/Sample comments are added two/i)).toBeTruthy();
    expect(within(video).getAllByText(/test provider one/i)).toHaveLength(2);
    expect(within(video).getAllByText(new RegExp(formatDate(observationList[0].encounterDateTime)))).toHaveLength(2);
    
    expect(container.querySelectorAll(".file-section .video-viewer")).toHaveLength(2);
    expect(container.querySelectorAll(".file-section .video-viewer.video-0")).toHaveLength(1);
    expect(container.querySelectorAll(".file-section .video-viewer.video-1")).toHaveLength(1);
    
    const image = fileSections[1];
    expect(image).not.toBeNull();

    expect(within(image).getByText(/Image/i)).toBeTruthy();
    expect(within(image).getByText(/Sample comments are added three/i)).toBeTruthy();
    expect(within(image).getByText(/Sample comments are added four/i)).toBeTruthy();
    expect(within(image).getAllByText(/test provider one/i)).toHaveLength(3);
    expect(within(image).getAllByText(new RegExp(formatDate(observationList[0].encounterDateTime)))).toHaveLength(3);
    
    expect(container.querySelectorAll(".file-section .image-viewer")).toHaveLength(3);
    expect(container.querySelectorAll(".file-section .image-viewer.img-0")).toHaveLength(1);
    expect(container.querySelectorAll(".file-section .image-viewer.img-1")).toHaveLength(1);
    expect(container.querySelectorAll(".file-section .image-viewer.img-2 .pdf-link")).toHaveLength(1);

  });

  it("should render and group complex type like image which is in second level of hierarchy", () => {
    const updatedProps = { ...initialProps, formData: observationListWithGroupMembers};
    const { container } = render(<ViewObservationForm {...updatedProps} />);

    const sectionHeader = container.querySelector(".section-header");
    expect(within(sectionHeader).getByText(/Consultation Images/i)).toBeTruthy();

    expect(container.querySelector(".row-label")).toBeTruthy();
    expect(container.querySelector(".row-label").textContent).toBe("Image 0-9-3");

    expect(container.querySelectorAll(".row-value .image-viewer")).toHaveLength(2);

    const imageRowOne = container.querySelector(".row-value .image-viewer.img-0");
    expect(imageRowOne).toBeTruthy();
    expect(within(imageRowOne).getByText(/Notes 1/i)).toBeTruthy();
    expect(within(imageRowOne).getByText(/provider one/i)).toBeTruthy();
    expect(within(imageRowOne).getByText(new RegExp(formatDate(observationListWithGroupMembers[0].encounterDateTime)))).toBeTruthy();

    const imageRowTwo = container.querySelector(".row-value .image-viewer.img-1");
    expect(imageRowTwo).toBeTruthy();
    expect(within(imageRowTwo).getByText(/Notes 2/i)).toBeTruthy();
    expect(within(imageRowTwo).getByText(/provider one/i)).toBeTruthy();
    expect(within(imageRowTwo).getByText(new RegExp(formatDate(observationListWithGroupMembers[0].encounterDateTime)))).toBeTruthy();
  });
});
