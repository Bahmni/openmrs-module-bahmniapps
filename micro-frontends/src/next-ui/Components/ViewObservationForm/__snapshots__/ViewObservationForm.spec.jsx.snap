// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewObservationForm should match the screenshot 1`] = `
<div>
  <div>
    <div
      class="bx--modal is-visible view-observation-form-modal"
      role="presentation"
    >
      <span
        class="bx--visually-hidden"
        role="link"
        tabindex="0"
      >
        Focus sentinel
      </span>
      <div
        aria-label=""
        aria-modal="true"
        class="bx--modal-container"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="bx--modal-header"
        >
          <button
            aria-label="close"
            class="bx--modal-close"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="bx--modal-close__icon"
              fill="currentColor"
              focusable="false"
              height="20"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 32 32"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
              />
            </svg>
          </button>
          
          <h3
            class="bx--modal-header__heading"
            id="bx--modal-header__heading--modal-1"
          />
        </div>
        <div
          aria-labelledby="bx--modal-header__heading--modal-1"
          class="bx--modal-content"
          id="bx--modal-body--modal-1"
        >
          <section
            class="content-body"
          >
            <h2
              class="section-title"
            >
              Vitals
            </h2>
            <section
              class="section-body"
            >
              <div
                class="bx--tile"
              >
                <div
                  style="display: flex;"
                >
                  <span
                    class="section-header row-label is-abnormal"
                    data-testid="section-label-0"
                  >
                    Pulse
                     
                    <span
                      class="sub-label"
                    >
                      (60 - 100)
                    </span>
                  </span>
                  <div
                    class="row-value is-abnormal"
                  >
                    105
                     
                    
                  </div>
                </div>
                <span
                  class="notes-section"
                >
                  <svg
                    aria-hidden="true"
                    class="document-icon"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M25.7,9.3l-7-7C18.5,2.1,18.3,2,18,2H8C6.9,2,6,2.9,6,4v24c0,1.1,0.9,2,2,2h16c1.1,0,2-0.9,2-2V10C26,9.7,25.9,9.5,25.7,9.3 z M18,4.4l5.6,5.6H18V4.4z M24,28H8V4h8v6c0,1.1,0.9,2,2,2h6V28z"
                    />
                    <path
                      d="M10 22H22V24H10zM10 16H22V18H10z"
                    />
                  </svg>
                  notes example - by test provider
                </span>
              </div>
              <div
                class="bx--tile"
              >
                <div
                  style="display: flex;"
                >
                  <span
                    class="section-header row-label "
                    data-testid="section-label-1"
                  >
                    Pulse
                     
                    <span
                      class="sub-label"
                    >
                      (60 - 100)
                    </span>
                  </span>
                  <div
                    class="row-value "
                  >
                    
                     
                    beats/min
                  </div>
                </div>
                <span
                  class="notes-section"
                >
                  <svg
                    aria-hidden="true"
                    class="document-icon"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M25.7,9.3l-7-7C18.5,2.1,18.3,2,18,2H8C6.9,2,6,2.9,6,4v24c0,1.1,0.9,2,2,2h16c1.1,0,2-0.9,2-2V10C26,9.7,25.9,9.5,25.7,9.3 z M18,4.4l5.6,5.6H18V4.4z M24,28H8V4h8v6c0,1.1,0.9,2,2,2h6V28z"
                    />
                    <path
                      d="M10 22H22V24H10zM10 16H22V18H10z"
                    />
                  </svg>
                  notes example - by 
                </span>
              </div>
              <div
                class="bx--tile"
              >
                <div
                  style="display: block;"
                >
                  <span
                    class="section-header  "
                    data-testid="section-label-2"
                  >
                    Blood Pressure
                     
                    <span
                      class="sub-label"
                    />
                  </span>
                  <div
                    class="row-body"
                  >
                    <div
                      class="row-element is-abnormal"
                    >
                      <div
                        class="row-content"
                      >
                        <div
                          class="row-label"
                        >
                          Systolic Blood Pressure
                           
                          <span
                            class="sub-label"
                          >
                            (100 - 140)
                          </span>
                        </div>
                        <div
                          class="row-value"
                        >
                          120
                           
                          mmHg
                        </div>
                      </div>
                      <span
                        class="notes-section"
                      >
                        <svg
                          aria-hidden="true"
                          class="document-icon"
                          fill="currentColor"
                          focusable="false"
                          height="16"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="16"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M25.7,9.3l-7-7C18.5,2.1,18.3,2,18,2H8C6.9,2,6,2.9,6,4v24c0,1.1,0.9,2,2,2h16c1.1,0,2-0.9,2-2V10C26,9.7,25.9,9.5,25.7,9.3 z M18,4.4l5.6,5.6H18V4.4z M24,28H8V4h8v6c0,1.1,0.9,2,2,2h6V28z"
                          />
                          <path
                            d="M10 22H22V24H10zM10 16H22V18H10z"
                          />
                        </svg>
                        notes example - by test provider
                      </span>
                    </div>
                    <div
                      class="row-element "
                    >
                      <div
                        class="row-content"
                      >
                        <div
                          class="row-label"
                        >
                          Diastolic Blood Pressure
                           
                          <span
                            class="sub-label"
                          >
                            (60 - 90)
                          </span>
                        </div>
                        <div
                          class="row-value"
                        >
                          80
                           
                          mmHg
                        </div>
                      </div>
                      <span
                        class="notes-section"
                      >
                        <svg
                          aria-hidden="true"
                          class="document-icon"
                          fill="currentColor"
                          focusable="false"
                          height="16"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="16"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M25.7,9.3l-7-7C18.5,2.1,18.3,2,18,2H8C6.9,2,6,2.9,6,4v24c0,1.1,0.9,2,2,2h16c1.1,0,2-0.9,2-2V10C26,9.7,25.9,9.5,25.7,9.3 z M18,4.4l5.6,5.6H18V4.4z M24,28H8V4h8v6c0,1.1,0.9,2,2,2h6V28z"
                          />
                          <path
                            d="M10 22H22V24H10zM10 16H22V18H10z"
                          />
                        </svg>
                        notes example - by 
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </section>
          </section>
        </div>
      </div>
      <span
        class="bx--visually-hidden"
        role="link"
        tabindex="0"
      >
        Focus sentinel
      </span>
    </div>
  </div>
</div>
`;
